libs/sdk-flutter/android/src/main/kotlin/com/breez/breez_sdk/BreezSDKPlugin.kt

```diff
@@ -1,35 +1,63 @@
 package com.breez.breez_sdk
 
-import androidx.annotation.NonNull
-
+import android.os.Handler
+import android.os.Looper
+import breez_sdk.LogEntry
 import io.flutter.embedding.engine.plugins.FlutterPlugin
+import io.flutter.plugin.common.EventChannel
 import io.flutter.plugin.common.MethodCall
 import io.flutter.plugin.common.MethodChannel
 import io.flutter.plugin.common.MethodChannel.MethodCallHandler
 import io.flutter.plugin.common.MethodChannel.Result
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
 
 /** BreezSDKPlugin */
-class BreezSDKPlugin: FlutterPlugin, MethodCallHandler {
-  /// The MethodChannel that will the communication between Flutter and native Android
-  ///
-  /// This local reference serves to register the plugin with the Flutter Engine and unregister it
-  /// when the Flutter Engine is detached from the Activity
-  private lateinit var channel : MethodChannel
-
-  override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
-    channel = MethodChannel(flutterPluginBinding.binaryMessenger, "breez_sdk")
-    channel.setMethodCallHandler(this)
-  }
-
-  override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
-    if (call.method == "getPlatformVersion") {
-      result.success("Android ${android.os.Build.VERSION.RELEASE}")
-    } else {
-      result.notImplemented()
+class BreezSDKPlugin : FlutterPlugin, MethodCallHandler, EventChannel.StreamHandler {
+    // Event has to be handled on main thread.
+    private var handler = Handler(Looper.getMainLooper())
+
+    private lateinit var channel: MethodChannel
+    private var eventChannel: EventChannel? = null
+    private var eventSink: EventChannel.EventSink? = null
+    private var scope = CoroutineScope(Dispatchers.Main)
+
+    override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
+        channel = MethodChannel(flutterPluginBinding.binaryMessenger, "breez_sdk")
+        channel.setMethodCallHandler(this)
+
+        eventChannel = EventChannel(flutterPluginBinding.binaryMessenger, "breez_sdk_node_logs")
+        val nodeLogStream = SdkLogInitializer.initializeNodeLogStream()
+        nodeLogStream.subscribe(scope) { l: LogEntry ->
+            val runnable = Runnable {
+                val data = mapOf("level" to l.level, "line" to l.line)
+                eventSink?.success(data)
+            }
+
+            handler.post(runnable)
```

Comment:
@erdemyerebasmaz  The subscribed scope seems to already use the main thread, can we omit the handler here and instead call directly the eventSink without the Runnable?

Reply:
Yes, good catch!
Done f1d89da3820aa17a37bb6d19d200d71559004df8